### PR TITLE
Refactor Quick Connect

### PR DESCRIPTION
**Pull request by nielsvanvelzen**
**Changes**

- Enable nullable in Quick Connect code
- Some miscellaneous refactoring in Quick Connect code  ,mostly based on Rider hints
- Increased Quick Connect timeout to 10 minutes (was 5)
- Move from 3-state (sort-of-on/on/off) to 2-state (on/off)
- Remove "activation" shit, when it's on in the server you can request codes. simple as that.
- Add proper HTTP responses when disabled (instead of HTTP 500)
- Change functions not used from interface to private / remove from interface
- Change token type to UUID
- Fix nullability in 
- Remove delete functions (unused)
- Read state directly from system config, this fixes updating the config not enabling quick connect

**API Changes**
- Removed "status" endpoint, replaced with "enabled" endpoint
  - Return type changed to boolean
- Add Unauthorized responses when Quick Connect is disabled
- Remove "activate" endpoint
- Remove "available" endpoint
  - Use system configuration api instead
- Remove "Deauthorize" endpoint

**Related pull requests**

jellyfin/jellyfin-web#2747
jellyfin/jellyfin-web#2748
jellyfin/jellyfin-web#2749

**Issues**

### CODE REVIEW

File: Emby.Server.Implementations/QuickConnect/QuickConnectManager.cs [hunk 1]
```diff
@@ -22,14 +19,28 @@ namespace Emby.Server.Implementations.QuickConnect
     /// </summary>
     public class QuickConnectManager : IQuickConnect, IDisposable
     {
-        private readonly RNGCryptoServiceProvider _rng = new RNGCryptoServiceProvider();
-        private readonly ConcurrentDictionary<string, QuickConnectResult> _currentRequests = new ConcurrentDictionary<string, QuickConnectResult>();
+        /// <summary>
+        /// Gets or sets the name of internal access tokens.
```
Comment (barronpm) [inline Emby.Server.Implementations/QuickConnect/QuickConnectManager.cs:23]:
```suggestion
        /// The name of internal access tokens.
```

File: Emby.Server.Implementations/QuickConnect/QuickConnectManager.cs [hunk 2]
```diff
@@ -22,14 +19,28 @@ namespace Emby.Server.Implementations.QuickConnect
...
+        /// </summary>
+        private const string TokenName = "QuickConnect";
+
+        /// <summary>
+        /// Gets or sets the length of user facing codes.
```
Comment (barronpm) [inline Emby.Server.Implementations/QuickConnect/QuickConnectManager.cs:28]:
```suggestion
        /// The length of user facing codes.
```

File: Emby.Server.Implementations/QuickConnect/QuickConnectManager.cs [hunk 3]
```diff
@@ -22,14 +19,28 @@ namespace Emby.Server.Implementations.QuickConnect
...
+        /// </summary>
+        private const int CodeLength = 6;
+
+        /// <summary>
+        /// Gets or sets the time (in minutes) before quick connect will automatically deactivate.
```
Comment (barronpm) [inline Emby.Server.Implementations/QuickConnect/QuickConnectManager.cs:33]:
```suggestion
        /// The time (in minutes) before quick connect will automatically deactivate.
```

File: Emby.Server.Implementations/QuickConnect/QuickConnectManager.cs [hunk 4]
```diff
@@ -22,14 +19,28 @@ namespace Emby.Server.Implementations.QuickConnect
...
+        /// </summary>
+        private const string TokenName = "QuickConnect";
+
+        /// <summary>
+        /// The length of user facing codes.
+        /// </summary>
+        private const int CodeLength = 6;
+
+        /// <summary>
+        /// The time (in minutes) before quick connect will automatically deactivate.
```
Comment (crobibero) [inline Emby.Server.Implementations/QuickConnect/QuickConnectManager.cs:33]:
```suggestion
        /// The time (in minutes) that the quick connect token is valid.
```

**Review by crobibero — APPROVED** [review]

**Review by BaronGreenback — APPROVED** [review]

File: Emby.Server.Implementations/QuickConnect/QuickConnectManager.cs [hunk 5]
```diff
@@ -22,14 +19,28 @@ namespace Emby.Server.Implementations.QuickConnect
...
+
+        /// <summary>
+        /// The length of user facing codes.
+        /// </summary>
+        private const int CodeLength = 6;
+
+        /// <summary>
+        /// The time (in minutes) that the quick connect token is valid.
+        /// </summary>
+        private const int Timeout = 10;
```
Comment (Bond-009) [inline Emby.Server.Implementations/QuickConnect/QuickConnectManager.cs:35]: We *could* easily make these configurable, possible future feature?

File: Emby.Server.Implementations/QuickConnect/QuickConnectManager.cs [hunk 6]
```diff
@@ -247,22 +195,16 @@ private string GenerateSecureRandom(int length = 32)
...
-                SetState(QuickConnectState.Available);
-                expireAll = true;
-            }
-
             // Expire stale connection requests
             foreach (var (_, currentRequest) in _currentRequests)
             {
-                var added = currentRequest.DateAdded ?? DateTime.UnixEpoch;
-                if (expireAll || DateTime.UtcNow > added.AddMinutes(Timeout))
+                if (expireAll || DateTime.UtcNow > currentRequest.DateAdded.AddMinutes(Timeout))
```
Comment (Bond-009) [inline Emby.Server.Implementations/QuickConnect/QuickConnectManager.cs:207]: `DateTime.UtcNow` is relatively expensive, maybe move it out of the loop.
(you can even subtract the `Timeout` outside of the loop then)

**Review by Bond-009 — APPROVED** [review]
Would love some tests for this ;)

**Review by cvium — APPROVED** [review]

File: Emby.Server.Implementations/QuickConnect/QuickConnectManager.cs (hunk 5 above)
Comment (nielsvanvelzen) [reply Emby.Server.Implementations/QuickConnect/QuickConnectManager.cs:35]: Sounds like a good feature, my goal for this PR was to just fix the current implementation so I can implement it in the Android TV app. I have absolutely no idea how I'd make this properly configurable :p

**Review by cvium — APPROVED** [review]

**Pull request merged**